backend/.air.toml

```diff
@@ -1,3 +1,6 @@
 [build]
+cmd = "go build -gcflags='all=-N -l' -o ./tmp/main ."
+bin = "tmp/main"
+full_bin = "dlv exec --accept-multiclient --log --headless --continue --listen :2345 --api-version 2 ./tmp/main"
```

Comment:
I got this setup from this [blog post](https://dev.to/bruc3mackenzi3/debugging-go-inside-docker-using-vscode-4f67). A couple things to mention:

* We install delve already in our backend Dockerfile ([source](https://github.com/highlight/highlight/blob/02c12243018162d6012853a21d14e5764d620c48/docker/backend.Dockerfile#L3))
* Port 2345 is what we run the debugger on ([source](https://github.com/highlight/highlight/blob/02c12243018162d6012853a21d14e5764d620c48/docker/compose.dev.yml#L15)
* VSCode is configured for remote debugging ([source](https://github.com/highlight/highlight/blob/02c12243018162d6012853a21d14e5764d620c48/.vscode/launch.json#L6-L15))

The one thing new is the `-gcflags='all=-N -l'` which per the blog post:

> We've added the flag -gcflags "all=-N -l" to the build command. This is to disable inlining and optimizations that can interfere with debugging.

I distinctly remember having a problem with our old setup whereby some variables would say something like "code is unreachable". I don't remember the exact message but I remember it happening when you were deep in a function. The only way around it was to manually print the variable name (super annoying).

From my brief testing, all variables are defined in the debugger regardless of where I set the breakpoint so this looks like a huge developer productivity boost:

![Screenshot 2023-03-20 at 9 56 06 AM](https://user-images.githubusercontent.com/58678/226397335-a8469a56-39dc-4985-a0dc-667be0b67433.png)